src/components/FilterFields/validators.js

```diff
@@ -0,0 +1,33 @@
+import moment from 'moment'
+import i18n from '@dhis2/d2-i18n'
+
+// The expected date format
+const expectedFormat = 'YYYY-MM-DD'
+
+// Validates two dates compared to eachother
+export const validateDateRange = (startDate, endDate) => {
+    const startMoment = moment(startDate, expectedFormat)
+    const endMoment = moment(endDate, expectedFormat)
+
+    if (startMoment.isAfter(endMoment)) {
+        return i18n.t('Start date should be before end date')
+    }
+
+    return ''
+}
+
+// Validates a single date
+export const validateDate = date => {
+    if (!date) {
+        return i18n.t('Please fill in a date')
+    }
+
+    // The 'true' flag ensures strict validation of the expected format
+    const dateMoment = moment(date, expectedFormat, true)
+
+    if (!dateMoment.isValid()) {
+        return i18n.t('Please use the format yyyy-mm-dd')
+    }
+
+    return ''
```

Comment:
The thing is, the value that's being returned here is being passed to components as a prop. Those components currently have required strings for that prop (which will error out if it's null or undefined).

I could remove the isRequired there, but then we lose the ability to detect whether it's receiving the validation message correctly. Which seems useful to have to me. And since we're not using final form I thought it ok to not conform to that convention here.